kivy/core/camera/camera_android.py

```diff
@@ -0,0 +1,156 @@
+
+        self._fbo = Fbo(size=self._resolution)
+        self._fbo.shader.fs = '''
+            #extension GL_OES_EGL_image_external : require
+            #ifdef GL_ES
+                precision highp float;
+            #endif
+
+            /* Outputs from the vertex shader */
+            varying vec4 frag_color;
+            varying vec2 tex_coord0;
+
+            /* uniform texture samplers */
+            uniform sampler2D texture0;
+            uniform samplerExternalOES texture1;
+
+            void main()
+            {
+                gl_FragColor = texture2D(texture1, tex_coord0);
+            }
+        '''
+        with self._fbo:
+            self._texture_cb = Callback(lambda instr: self._camera_texture.bind)
+            Rectangle(size=self._resolution)
+
+    def _on_preview_frame(self, data, camera):
+        with self._buflock:
+            if self._buffer is not None:
+                self._android_camera.addCallbackBuffer(self._buffer)  # add buffer back for reuse
+            self._buffer = data
+        #print self._buffer, len(self.frame_data)  # check if frame grabbing works
+
+    def _refresh_fbo(self):
+        self._texture_cb.ask_update()
+        self._fbo.draw()
+
+    def start(self):
+        super(CameraAndroid, self).start()
+
+        with self._buflock:
+            self._buffer = None
+        for k in range(2):  # double buffer
+            buf = '\x00' * self._bufsize
+            self._android_camera.addCallbackBuffer(buf)
+        self._android_camera.setPreviewCallbackWithBuffer(self._preview_cb)
+
+        self._android_camera.startPreview()
+        Clock.unschedule(self._update)
+        Clock.schedule_interval(self._update, 1./self.fps)
+
+    def stop(self):
+        super(CameraAndroid, self).stop()
+        Clock.unschedule(self._update)
+        self._android_camera.stopPreview()
+
+        self._android_camera.setPreviewCallbackWithBuffer(None)  # buffer queue cleared as well, to be recreated on next start
+        with self._buflock:
+            self._buffer = None
+
+    def _update(self, dt):
+        self._surface_texture.updateTexImage()
+        self._refresh_fbo()
+        if self._texture is None:
+            self._texture = self._fbo.texture
+            self.dispatch('on_load')
+        self._copy_to_gpu()
+
+    def _copy_to_gpu(self):
+        """
+        A dummy placeholder (the image is already in GPU) to be consistent with other providers.
+        """
+        self.dispatch('on_texture')
+
+    def grab_frame(self):
+        """
+        Grab current frame (thread-safe, minimal overhead)
+        """
+        with self._buflock:
+            if self._buffer is None:
+                return None
+            buf = self._buffer.tostring()
+            return buf
+
+    def decode_frame(self, buf):
+        """
+        Decode image data from grabbed frame
+        """
+        import numpy as np
+        from cv2 import cvtColor
```

Comment:
This makes numpy and cv2 as a essential requirement on android. Either we should look at replacing these or making bumpy and cv2 as a base requirement for `kivy` recipe in py4a. I assume these are used for performance reasons.

Would it be possible to replace these with another implementation that is equally fast but does not require loading these libs?